plugins/backend/decsync/decsyncInterface.vala

```diff
@@ -0,0 +1,738 @@
+	{
+		return false;
+	}
+
+	public bool doInitSync()
+	{
+		return false;
+	}
+
+	public string symbolicIcon()
+	{
+		return "feed-service-decsync-symbolic";
+	}
+
+	public string accountName()
+	{
+		return "DecSync";
+	}
+
+	public string getServerURL()
+	{
+		return "http://localhost/";
+	}
+
+	public string uncategorizedID()
+	{
+		return "0";
+	}
+
+	public bool hideCategoryWhenEmpty(string catID)
+	{
+		return false;
+	}
+
+	public bool supportCategories()
+	{
+		return true;
+	}
+
+	public bool supportFeedManipulation()
+	{
+		return true;
+	}
+
+	public bool supportMultiLevelCategories()
+	{
+		return true;
+	}
+
+	public bool supportMultiCategoriesPerFeed()
+	{
+		return false;
+	}
+
+	public bool syncFeedsAndCategories()
+	{
+		return false;
+	}
+
+	public bool tagIDaffectedByNameChange()
+	{
+		return false;
+	}
+
+	public void resetAccount()
+	{
+		return;
+	}
+
+	public bool useMaxArticles()
+	{
+		return true;
+	}
+
+	public LoginResponse login()
+	{
+		if (initDecsync())
+		{
+			return LoginResponse.SUCCESS;
+		}
+		else
+		{
+			return LoginResponse.ALL_EMPTY;
+		}
+	}
+
+	public bool logout()
+	{
+		return true;
+	}
+
+	public bool serverAvailable()
+	{
+		return Utils.ping("https://duckduckgo.com/");
+	}
+
+	public void setArticleIsRead(string articleIDs, ArticleStatus readStatus)
+	{
+		var read = readStatus == ArticleStatus.READ;
+		Logger.debug("Mark " + articleIDs + " as " + (read ? "read" : "unread"));
+		var entries = new Gee.ArrayList<Decsync.EntryWithPath>();
+		foreach (var articleID in articleIDs.split(","))
+		{
+			Article? article = m_db.read_article(articleID);
+			if (article != null)
+			{
+				var path = articleToPath(article, "read");
+				var key = stringToNode(article.getArticleID());
+				entries.add(new Decsync.EntryWithPath.now(path, key, boolToNode(read)));
+			}
+		}
+		m_sync.setEntries(entries);
+	}
+
+	public void setArticleIsMarked(string articleID, ArticleStatus markedStatus)
+	{
+		var marked = markedStatus == ArticleStatus.MARKED;
+		Logger.debug("Mark " + articleID + " as " + (marked ? "marked" : "unmarked"));
+		Article? article = m_db.read_article(articleID);
+		if (article != null)
+		{
+			var path = articleToPath(article, "marked");
+			var key = stringToNode(article.getArticleID());
+			m_sync.setEntry(path, key, boolToNode(marked));
+		}
+	}
+
+	public bool alwaysSetReadByID()
+	{
+		return true;
+	}
+
+	public void setFeedRead(string feedID)
+	{
+		return;
+	}
+
+	public void setCategoryRead(string catID)
+	{
+		return;
+	}
+
+	public void markAllItemsRead()
+	{
+		return;
+	}
+
+	public void tagArticle(string articleID, string tagID)
+	{
+		return;
+	}
+
+	public void removeArticleTag(string articleID, string tagID)
+	{
+		return;
+	}
+
+	public string createTag(string caption)
+	{
+		string tagID = "1";
+
+		if(!m_db.isTableEmpty("tags"))
+			tagID = (int.parse(m_db.getMaxID("tags", "tagID")) + 1).to_string();
+
+		Logger.info("createTag: ID = " + tagID);
+		return tagID;
```

Comment:
I'm confused about this function since `supportTags()` is false. This function should probably either return an empty string or throw an exception.

(At some point I want to redo how plugins work so these functions don't even exist unless the plugin advertises supporting them, but for now just not having any logic in them is good enough)